core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -132,47 +134,40 @@ public int read(byte[] b, int off, int len) throws IOException {
     if (mPosition >= mStatus.getLength()) { // at end of file
       return -1;
     }
-    int bytesRead = 0;
+    int totalBytesRead = 0;
     long lengthToRead = Math.min(len, mStatus.getLength() - mPosition);
     // for each page, check if it is available in the cache
-    while (bytesRead < lengthToRead) {
+    while (totalBytesRead < lengthToRead) {
       long currentPage = mPosition / mPageSize;
       int currentPageOffset = (int) (mPosition % mPageSize);
-      int bytesLeftInPage = (int) Math.min(mPageSize - currentPageOffset, lengthToRead - bytesRead);
+      int bytesLeftInPage =
+          (int) Math.min(mPageSize - currentPageOffset, lengthToRead - totalBytesRead);
       PageId pageId = new PageId(mStatus.getFileIdentifier(), currentPage);
-      try (ReadableByteChannel cachedData = mCacheManager.get(pageId, currentPageOffset)) {
```

Comment:
move the IO operations inside `LocalCacheManager` under page locks to coordinate with other threads, also procted by `NoExceptionManager`